ocp_resources/catalog_source.py

```diff
@@ -10,13 +11,13 @@ class CatalogSource(NamespacedResource):
 
     def __init__(
         self,
-        source_type=None,
-        image=None,
-        display_name=None,
-        publisher=None,
-        update_strategy_registry_poll_interval=None,
-        **kwargs,
-    ):
+        source_type: str = "",
```

Comment:
```suggestion
        source_type: str | None = None,
```

Reply:
Empty str is OK here, we check if this is not empty.